Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -818,20 +818,18 @@ RefPtr<StyleRuleFontPaletteValues> CSSParserImpl::consumeFontPaletteValuesRule(C
             if (value.isFontFamily())
                 fontFamilies.append(AtomString { value.stringValue() });
         };
-        auto cssFontFamily = properties->getPropertyCSSValue(CSSPropertyFontFamily);
+        RefPtr cssFontFamily = properties->getPropertyCSSValue(CSSPropertyFontFamily);
         if (!cssFontFamily)
             return fontFamilies;
-        auto cssPrimitiveFontFamilies = dynamicDowncast<CSSValueList>(*cssFontFamily);
-        if (cssPrimitiveFontFamilies) {
+        if (RefPtr cssPrimitiveFontFamilies = dynamicDowncast<CSSValueList>(*cssFontFamily)) {
             for (auto& item : *cssPrimitiveFontFamilies)
                 append(downcast<CSSPrimitiveValue>(item));
             return fontFamilies;
         }
-
-        auto cssPrimitiveFontFamily = dynamicDowncast<CSSPrimitiveValue>(*cssFontFamily);
-        if (!cssPrimitiveFontFamily)
+        if (RefPtr cssPrimitiveFontFamily = dynamicDowncast<CSSPrimitiveValue>(*cssFontFamily)) {
+            append(*cssPrimitiveFontFamily);
             return fontFamilies;
```

Comment:
It doesn't seem like this early return is needed with your new code path, since it should just fall through to returning `fontFamilies`.